pkg/install/install.go

```diff
@@ -162,6 +163,7 @@ func (i *Installer) AdminUpgrade(ctx context.Context) error {
 		action(i.configureAPIServerCertificate),
 		action(i.configureIngressCertificate),
 		action(i.upgradeCluster),
+		action(i.addResourceProviderVersion), // Run this last so we capture the resource provider only once the upgrade has been fully performed
```

Comment:
I think we need to cover `runSteps` with tests. One of the tests should make sure that if one step (action/condition) fails - we do not proceed with the next ones.

What do you think? Maybe not in this PR (create an issue for it?)

Reply:
I agree, but not this PR.  I think we should pull runSteps out into a separate package as I think I'd like to start using it in delete as well.  https://github.com/Azure/ARO-RP/issues/856.